libraries/GCS_MAVLink/MAVLink_routing.cpp

```diff
@@ -101,11 +101,6 @@ bool MAVLink_routing::check_and_forward(mavlink_channel_t in_channel, const mavl
     // so that find_mav_type works for all channels
     learn_route(in_channel, msg);
 
-    // don't ever forward data from a private channel
-    if ((GCS_MAVLINK::is_private(in_channel))) {
```

Comment:
this causes handle_heartbeat() to resend private chan heartbeats to all channels

Reply:
It doesn't, actually.  I'd actually already looked at that previously and noted that `handle_heartbeat` is private-channel-aware - just forgot on the DevCall.  The first thing it does is mask out private channels from the things it might forward to.

I've added a pair of patches on top of this PR to make that fact more visible.

Reply:
its the src check not the dest check that matters

Reply:
Fixed.